### ISSUE TITLE
Should entity_plus_type just be entity_type?

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/entity_plus#44]
This looks like potentially the result of an overeager search and replace at some point... @hosef can you verify the PR that's coming shortly? I think there are a bunch of references to `entity_plus_type` that should just be `entity_type`...

**Comment by hosef** [comment]
@laryn I took a look at the PR and those all seem like reasonable changes. I agree that we should be using entity instead of entity_plus whenever possible.

**Comment by argiepiano** [comment]
@hosef and @laryn : since this change involves mostly variables, I agree that there are no issues there. There are a couple of instances where the change has already been merged (see #83 , where 'entity_plus_type' threw an error when passed to EntityFieldQuery.).

**Comment by laryn** [comment]
Great -- thanks both for looking this over. Feels cleaner to just use `$entity_type`!

**Comment by argiepiano** [comment]
@laryn, thanks for creating an updated PR (#151). Unlike the old one, the new PR, as you said on Zulip, only changes local variables ($entity_plus_type to $entity_type) and therefore is very safe. 

The old PR contained two changes that are not variables (those did not make it into the new PR):

- In `EntityPlusController::buildContent()`, there is an instance of an array key `entity_plus_type` (line 641 of `entity_plus.controller.inc`), which your old PR changed to `entity_type`. This is in the $context array, which is sent to implementations of `hook_entity_plus_view_mode_alter()`. (the equivalent hook in core is called `hook_entity_view_mode_alter()`). It's likely that other ported contrib modules ended up using the core version of this alter hook. If any module actually used E+'s  `hook_entity_plus_view_mode_alter()`, it's likely they have already switched to expecting `entity_plus_type` in the $context array, since this has been like this for a long time. So we could do two things in this case:
  1. Leave this alone and not change it (which would prevent breaking things in contrib). This is basically what your new PR does. 
  2. OR, actually send the entity type keyed under BOTH, `entity_plus_type` and `entity_type` just to prevent breaking things, but at the same time provide for any possible implementations that still expect the entity being keyed under `entity_type` 

- The OTHER instance of a non-variable change in your old PR was  in `EntityPlusController::view()`, in the render array `$build`. In that case, that render array (which uses the theme `'entity_plus'` as defined under `entity_plus_theme()`) sent the entity type keyed under `#entity_plus_type` . Your old PR changed that to `#entity_type`. The good news is that you also changed the preprocess function `template_preprocess_entity_plus()` to expect the type to appear under `$variables['entity_type']`. But, like above, the problem of changing this is that there may be contrib overrides of this template function that expect `entity_plus_type`.  So, like above, we have to options
  1. Not mess with it and leave `#entity_plus_type` intact, as in your new PR 
  2. OR, provide the entity type keyed under both `#entity_type_plus` and `#entity_type`.

So, in short, your new PR looks fine, and safer than the old PR, as it leaves those two instances alone, and therefore has fewer chances of breaking newer implementations. 

Of course, with Entity Plus, it's always hard to decide if we want to preserve things as they are now to prevent breaking things that have been changed since the E+ Backdrop version was released, or if we want to do things to make it easier for people to port modules that used Entity API.

**Comment by laryn** [comment]
Thanks for reviewing @argiepiano and @hosef -- I've added a few more commits on the new PR for additional review. I thought perhaps best to standardize these other instances and have also added the backwards-compatible duplications you suggested, @argiepiano. (I notice for example, that the more recent work to add entity property theming uses the standard `#entity_type` key, and it makes sense to keep things standard and consistent, I think)

> Of course, with Entity Plus, it's always hard to decide if we want to preserve things as they are now to prevent breaking things that have been changed since the E+ Backdrop version was released, or if we want to do things to make it easier for people to port modules that used Entity API.

Agreed!

**Comment by argiepiano** [comment]
Thanks for the new changes in the PR. LGTM and RTBC!

**Comment by hosef** [comment]
@laryn I think you should also change the comments here https://github.com/backdrop-contrib/entity_plus/blob/1.x/theme/entity_plus.tpl.php#L19.

Other than that, I think this is good to go. I ran tests on my local environment and made sure nothing was broken when using modules that depend on entity plus.

**Comment by laryn** [comment]
Thanks @argiepiano and @hosef -- latest changes made.

**Comment by argiepiano** [comment]
Looks good to me! While looking into this PR, I caught an issue with the template documentation, and how E+ passes css classes to the template. I opened a separate issue for that. #152
